internal/core/src/index/JsonInvertedIndex.cpp

```diff
@@ -33,19 +33,23 @@ JsonInvertedIndex<T>::build_index_for_json(
     for (const auto& data : field_datas) {
         auto n = data->get_num_rows();
         for (int64_t i = 0; i < n; i++) {
-            auto json_column = static_cast<const Json*>(data->RawValue(i));
             if (this->schema_.nullable() && !data->is_valid(i)) {
                 this->null_offset.push_back(i);
+                this->wrapper_->template add_array_data<T>(
+                    nullptr, 0, offset++);
                 continue;
             }
+            auto json_column = static_cast<const Json*>(data->RawValue(i));
```

Comment:
do we need to cache the parser result per row?

right now we are iterating per field then per row.
if we iterate per row then per field does it performs better?

Reply:
> do we need to cache the parser result per row?
> 
> right now we are iterating per field then per row. if we iterate per row then per field does it performs better?

we are iterating the same json field here. all `fields_data` are data of this field.